lib/membrane_vpx/decoder/vpx_decoder.ex

```diff
@@ -51,51 +53,24 @@ defmodule Membrane.VPx.Decoder do
   @spec handle_buffer(:input, Membrane.Buffer.t(), CallbackContext.t(), State.t()) ::
           callback_return()
   def handle_buffer(:input, %Buffer{payload: payload, pts: pts}, ctx, state) do
-    {:ok, [decoded_frame], pixel_format} = Native.decode_frame(payload, state.decoder_ref)
+    {:ok, [decoded_frame]} = Native.decode_frame(payload, state.decoder_ref)
 
-    stream_format_action =
-      if ctx.pads.output.stream_format == nil do
-        output_stream_format =
-          get_output_stream_format(ctx.pads.input.stream_format, pixel_format, state)
+    new_stream_format = %RawVideo{
+      width: decoded_frame.width,
+      height: decoded_frame.height,
+      framerate: state.framerate,
+      pixel_format: decoded_frame.pixel_format,
+      aligned: true
+    }
 
-        [stream_format: {:output, output_stream_format}]
+    stream_format_action =
```

Comment:
Let's raise or warn if the frame shape get from native code is different than the shape defined in input stream format (if it has any information about it)